packages/agent/src/certificate.ts

```diff
@@ -101,57 +100,89 @@ function isBufferEqual(a: ArrayBuffer, b: ArrayBuffer): boolean {
 
 export class Certificate {
   private readonly cert: Cert;
-  private verified = false;
-  private _rootKey: ArrayBuffer | null = null;
 
-  constructor(response: ReadStateResponse, private _agent: Agent = getDefaultAgent()) {
-    this.cert = cbor.decode(new Uint8Array(response.certificate));
+  /**
+   * Create a new instance of a certificate, automatically verifying it.
+   * @throws {CertificateVerificationError}
+   */
+  public static async create(
```

Comment:
For constructors, we prefer to use a single options object instead of positional arguments. Do you mind pulling these out into a new type, and then documenting the object's properties in the JSDOC comment? See https://github.com/dfinity/agent-js/blob/main/packages/auth-client/src/index.ts#L212 as a reference